packages/toolpad-app/src/server/data.ts

```diff
@@ -376,7 +377,14 @@ export async function execApi<Q>(
       `Unknown connection "${api.attributes.connectionId.value}" for api "${api.id}"`,
     );
   }
-
+  if (api.attributes.transformEnabled?.value) {
+    const apiResult = await dataSource.exec(connection, api.attributes.query.value, params);
```

Comment:
To express the logic more clearly, perhaps extract it out of the `if` statement like:

```tsx
let result  = await exec()
if (transform) {
  result = transform(result)
}
return result;
```